### PR TITLE
fix v1 validation return error

### LINKED ISSUE #51304: Errors returned by some validation functions in api/v1/validation may be wrong
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
func  validateContainerResourceName and validateResourceName return field.ErrorList{} at last may be wrong 
**What you expected to happen**:
return correctly
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #51304 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v1/validation/validation.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func validateContainerResourceName(value string, fldPath *field.Path) field.Erro
 			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
 		}
 	}
-	return field.ErrorList{}
+	return allErrs
```
Comment (xiangpengzhao) [inline pkg/api/v1/validation/validation.go]: In fact, all errors are returned in the above return(s). But here both are ok, return allErrs might be safer.

Comment (hzxuzhonghu) [reply pkg/api/v1/validation/validation.go]: yes, by the way, there are many validations, which do not return until all fields are validated.  (https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/validation/validation.go#L131) etc.
should we return err ASAP or after all validated. To improve apiserver performance, I think return err ASAP is better.

Comment (NickrenREN) [reply pkg/api/v1/validation/validation.go]: I do not think all errs are returned.  validateResourceName(...) may return errs but when `if len(strings.Split(value, "/")) != 1` , nothing will be returned.  That is not expected.

Comment (NickrenREN) [reply pkg/api/v1/validation/validation.go]: I sent an issue and pr for this before.  But do not consider v1 package.
https://github.com/kubernetes/kubernetes/pull/51257
https://github.com/kubernetes/kubernetes/issues/51256

Comment (hzxuzhonghu) [reply pkg/api/v1/validation/validation.go]: original code 
```
func validateContainerResourceName(value string, fldPath *field.Path) field.ErrorList {
	allErrs := validateResourceName(value, fldPath)
	if len(strings.Split(value, "/")) == 1 {
		if !helper.IsStandardContainerResourceName(value) {
			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
		}
	}
	return field.ErrorList{}
}
```
as @NickrenREN  say, when ```len(strings.Split(value, "/")) != 1```  may return wrong result.

Comment (xiangpengzhao) [reply pkg/api/v1/validation/validation.go]: Good point! I misread :)

**Comment by hzxuzhonghu** [comment]
ping @davidopp @NickrenREN @thockin any else advice ?

**Comment by m1093782566** [comment]
I like this fix. LGTM

**Comment by hzxuzhonghu** [comment]
@m1093782566 thanks

**Comment by hzxuzhonghu** [comment]
@thockin  thanks

**Pull request merged**